commit 867b599b37e0
Author: Benjamin Bouvier <benj@benj.me>
Date:   Tue Oct 3 19:57:05 2017 +0200

    Bug 1404714 - inIon() should return false when in wasm. r=nbp, a=test-only on a CLOSED TREE
    
    MozReview-Commit-ID: 76qU9UjcROt
    
    --HG--
    extra : source : 6956f4fcbb7973e93c7314eb8bb9e7f8de93b96f
    extra : amend_source : ed03aa30810d829dbf0951ccbc7aaab805afa280
    extra : intermediate-source : d5f57ad0a0fb5b362ed6abc150aa1038a1555fa5
---
 js/src/builtin/TestingFunctions.cpp | 7 +++++++
 1 file changed, 7 insertions(+)

diff --git js/src/builtin/TestingFunctions.cpp js/src/builtin/TestingFunctions.cpp
index cf2f0e3ecc2a..5cc776d0a7cc 100644
--- js/src/builtin/TestingFunctions.cpp
+++ js/src/builtin/TestingFunctions.cpp
@@ -2199,6 +2199,13 @@ testingFunc_inIon(JSContext* cx, unsigned argc, Value* vp)
         return true;
     }
 
+    if (cx->activation()->isWasm()) {
+        // Exited through wasm. Note this is false when the fast wasm->jit exit
+        // was taken, in which case we actually have jit frames on the stack.
+        args.rval().setBoolean(false);
+        return true;
+    }
+
     ScriptFrameIter iter(cx);
     if (!iter.done() && iter.isIon()) {
         // Reset the counter of the IonScript's script.
